### PR TITLE
SOLR-14385 added shard name thru MDC and collection name to split histogram logs

### PR DESCRIPTION
**Pull request by saatchibhalla**
https://issues.apache.org/jira/browse/SOLR-14385

# Description

Please provide a short description of the changes you're making with this pull request.
Added shard name and collection name to split histogram logs. 

# Solution

Please provide a short description of the approach taken to implement your solution.
Had access to collection and now accessing shard name directly from MDC.

# Tests

Please describe the tests you've developed or run to confirm this patch implements the feature or solves the problem.
Modified SplitHandlerTest file to include creation of a test cluster with test collection to verify logs. Verified all related tests pass.

# Checklist

Please review the following and check all that apply:

- [x] I have reviewed the guidelines for [How to Contribute](https://wiki.apache.org/solr/HowToContribute) and my code conforms to the standards described there to the best of my ability.
- [x] I have created a Jira issue and added the issue ID to my pull request title.
- [ ] I have given Solr maintainers [access](https://help.github.com/en/articles/allowing-changes-to-a-pull-request-branch-created-from-a-fork) to contribute to my PR branch. (optional but recommended)
- [x] I have developed this patch against the `master` branch.
- [ 01894589] I have run `ant precommit` and the appropriate test suite.
- [x] I have added tests for my changes.
- [ ] I have added documentation for the [Ref Guide](https://github.com/apache/lucene-solr/tree/master/solr/solr-ref-guide) (for Solr changes only).

### CODE REVIEW

File: solr/core/src/java/org/apache/solr/handler/admin/SplitOp.java [hunk 1]
```diff
@@ -26,6 +26,7 @@
 import java.util.Map;
 import java.util.TreeMap;
 
+import org.apache.log4j.MDC;
```
Comment (dsmiley) [inline solr/core/src/java/org/apache/solr/handler/admin/SplitOp.java:29]: This is the wrong MDC; see MDCLoggingContext which imports `org.slf4j.MDC`

File: solr/core/src/test/org/apache/solr/handler/admin/SplitHandlerTest.java [hunk 2]
```diff
@@ -22,24 +22,50 @@
...
+  private static String systemCollNode;
+  private static String COLL_NAME = "collection1";
+  private static String SHARD_NAME = "shard1";
+  
   @BeforeClass
   public static void beforeTests() throws Exception {
     System.setProperty("managed.schema.mutable", "true");  // needed by cloud-managed config set
     initCore("solrconfig.xml","schema_latest.xml");
+    
+    configureCluster(2)
```
Comment (dsmiley) [inline solr/core/src/test/org/apache/solr/handler/admin/SplitHandlerTest.java:56]: This seems wrong given this test calls initCore and thus isn't a SolrCloud test.  Maybe it needs to _become_ a SolrCloud test?  On the other hand, it's nice to keep a test lightweight; the only motivation here is for logging?  Doesn't seem justified to me IMO.

CC @yonik

**Comment by dsmiley** [comment]
I'm unsure this is needed at all.  See my comment on JIRA.  Basically, doesn't the replica name already in MDC effectively tell you what you want to know?

File: solr/core/src/test/org/apache/solr/handler/admin/SplitHandlerTest.java (hunk 2 above)
Comment (megancarey) [reply solr/core/src/test/org/apache/solr/handler/admin/SplitHandlerTest.java:56]: That's fair @dsmiley - do you know if a way to create a mock collection so we can leave as SolrTestCaseJ4 but still avoid NPE from the new log line?

Comment (dsmiley) [reply solr/core/src/test/org/apache/solr/handler/admin/SplitHandlerTest.java:56]: No.  Maybe some _other_ test that is _already_ SolrCloud based could assert whatever log statement you want to do?

IMO log statements don't _need_ tests.  A nice to have.  If it's hard, spend your engineering hours on more worthwhile things.

**Pull request closed without merging**